src/main/java/org/graylog/collector/file/GlobPathSet.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.collector.file;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class GlobPathSet implements PathSet {
+    private static final Logger LOG = LoggerFactory.getLogger(GlobPathSet.class);
+
+    private final String pattern;
+    private final FileTreeWalker fileTreeWalker;
+    private final PathMatcher matcher;
+    private final Path rootPath;
+
+    public static class GlobbingFileVisitor extends SimpleFileVisitor<Path> {
+        private final PathMatcher matcher;
+        private final ImmutableSet.Builder<Path> matchedPaths;
+
+        public GlobbingFileVisitor(PathMatcher matcher, ImmutableSet.Builder<Path> matchedPaths) {
+            this.matcher = matcher;
+            this.matchedPaths = matchedPaths;
+        }
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+            // Skip /proc because it can throw some permission errors we cannot check for.
+            return dir.toString().startsWith("/proc") ? FileVisitResult.SKIP_SUBTREE : FileVisitResult.CONTINUE;
```

Comment:
This means that for example `/procurement` will also be skipped? Maybe better check for equality with `/proc` *or* prefix `/proc/`.